bindings/java/src/lib.rs

```diff
@@ -285,3 +356,39 @@ fn jstring_to_string(env: &mut JNIEnv, s: &JString) -> Result<String> {
     let res = unsafe { env.get_string_unchecked(s)? };
     Ok(res.into())
 }
+
+fn metakey_to_flagset(env: &mut JNIEnv, metakeys: jintArray) -> Result<Option<FlagSet<Metakey>>> {
```

Comment:
Please note that bindings should not map core API directly. We should provide API that native and friendly in Java. 

Will java users happy with API like list_with and metakey?

Reply:
Should we provide all `Metakey` by default?

Reply:
> Should we provide all Metakey by default?

I suggest to think about how will we provide those features without the knowledge of OpenDAL. And then make our decisions.

For example, how will a java library design an API that may have different options?